src/conf_mode/system-ipv6.py

```diff
@@ -69,12 +70,19 @@ def generate(ip_opt):
 def apply(ip_opt):
     # disable IPv6 address assignment
     if ip_opt['disable_addr_assignment']:
+        # this file is useless, but it's an indicator for grub setup scripts
         with open(ipv6_disable_file, 'w') as f:
             f.write('options ipv6 disable_ipv6=1')
+
+        if os.path.exists(grub_cfg_file):
+            os.system("sed -i 's/noautologin \+net.ifnames/noautologin ipv6.disable=1 net.ifnames/' "+grub_cfg_file)
```

Comment:
I think this should be more robust to changes in the cmdline, if the cmdline parameters are changed in the future, this will stop working. Perhaps append it to the end of the line and search for some other string that's more permanent.

Reply:
For example 
```
sed -i '/[[:space:]]*linux / s/$/ ipv6.disable=1/'
```
and replace the other places with matching code

Reply:
```
if ! sed -i 's/ipv6.disable=?/ipv6.disable=1/' $file; then
    sed -i '/[[:space:]]*linux / s/$/ ipv6.disable=1/' $file
fi
```